contributors/design-proposals/live-and-inplace-vertical-scaling.md

```diff
@@ -0,0 +1,325 @@
+Also, the 'API server' restores the resource requirements of each container of the PodSpec to the original and writes the revised PodSpec to ETCD to communicate with the Scheduler.
+This is because at this moment the PodSpec on ETCD shouldn’t be updated with new resource requirements.
+
+For a pod with ResizeRequested, the 'Scheduler' checks if the node on which the pod currently runs has enough resources to resize the pod.
+The Scheduler notify the 'API server' of the result via 'Resizing' API operation, which will be describe below.
+
+* **Resizing**
+
+A new API, Resizing, for the 'scheduler' is introduced:
+
+```go
+// Resizing resizes the resources allocated to a pod
+type Resizing struct {
+    metav1.TypeMeta
+    metav1.ObjectMeta
+    Request ResizeRequest
+}
+```
+
+Resizing has the metadata of a pod to resize and a value of ResizeRequest that holds the status of a resizing request, which indicates whether the resizing is feasible or not, and new resource requirements of the pod.
+
+Once the 'Scheduler' determine whether a resource resizing on a pod is feasible, or not, it notifies to the API server via this Resizing API.
+ObjectMeta.Namespace/Name are set to the Namespace and Name of the pod to resize and ResizeRequest is set with new resource requirements.
+
+Given a Resizing API operation, the ResizeStatus of the PodSpec of a Pod is updated according to that of the Resizing operation.
+If the ResizeStatus is ResizeAccepted, the API server updates the ResourceRequirement of each container of a pod with new resource requirements on ETCD.
+
+* **PodResized**
+
+A new pod condition, PodResized, and condition status for that is added to v1.PodCondition:
+
+```go
+// These are valid conditions of pod.
+const (
+       // PodResized represents the status of the resizing process for this pod
+    PodResized PodConditionType = "PodResized"
+    PodReasonUnresizable   = "Unresizable"
+    PodReasonResizerFailed = "ResizerFailed"
+)
+const (
+    ConditionRequested ConditionStatus = "Requested"
+    ConditionAccepted  ConditionStatus = "Accepted"
+    ConditionRejected  ConditionStatus = "Rejected"
+    ConditionDone      ConditionStatus = "Done"
+)
+```
+
+The pod condition of PodResized represents the status of the resizing process.
+The PodResized Condition is updated by the `Kubelet` according to the ResizeStatus, which is updated by the 'API server'.
+
+Basically, when the ResizeStatus is changed, the `Kubelet` updates the PodResized condition accordingly.
+In case of ConditionDone, the `Kubelet` sets the PodResized of a Pod to it when all the containers that need to be resized complete to be resized.
+
+
+* **A new additional hash, called `expectedHashNoResources`, added for `Kubelet` to detect a change on resource requirements**
+
+In order to watch resource requirement changes efficiently, a new additional hash is added to kubecontainer.ContainerStatus (and that is also stored as a one of the container’s labels).
```

Comment:
I am confused here.  It feels like we are trying to combine two pieces of information in a single field, and this hash is the outcome.  We are basically saying we need a desired resource requirement (what we have today and what we are looking to change) and an actual enforced resource requirement which kubelet acknowledged.

Reply:
Uses of persisted hashes have proven problematic between releases and in the face of version skew. Are we sure we want to add another one?

Reply:
I understand that the original hash is introduced to detect the pod spec changes in an efficient way, like avoiding one-by-one field comparisons. So, I want to keep this behavior, meaning not to cause additional overhead in pod spec changes detection even when the resource requirements in a pod spec is mutable. So, basically, with this new hash, the changes of other fields in a pod spec can be detected by simple has comparison.